internal/plugin/service/service/models.go

```diff
@@ -0,0 +1,144 @@
+package service
+
+import (
+	"context"
+
+	"github.com/aiven/aiven-go-client/v2"
+	"github.com/docker/go-units"
+	"github.com/hashicorp/terraform-plugin-framework-timeouts/resource/timeouts"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+// Plugin framework doesn't support embedded structs
+// https://github.com/hashicorp/terraform-plugin-framework/issues/242
+// We use resource as a base model, and copy state to/from dataSourceModel for datasource
+
+type Resource struct {
+	Timeouts              timeouts.Value `tfsdk:"timeouts"`
+	ID                    types.String   `tfsdk:"id"`
+	AdditionalDiskSpace   types.String   `tfsdk:"additional_disk_space"`
+	CloudName             types.String   `tfsdk:"cloud_name"`
+	Components            types.Set      `tfsdk:"components"`
+	DiskSpaceCap          types.String   `tfsdk:"disk_space_cap"`
+	DiskSpaceDefault      types.String   `tfsdk:"disk_space_default"`
+	DiskSpaceStep         types.String   `tfsdk:"disk_space_step"`
+	DiskSpaceUsed         types.String   `tfsdk:"disk_space_used"`
+	Kafka                 types.Set      `tfsdk:"kafka"`
```

Comment:
nit: if this is meant to represent Kafka configurations, consider renaming to something like `KafkaConfigs` for clarity